src/Approximations/template_directions.jl

```diff
@@ -374,8 +407,80 @@ SphericalDirections(Nθ::Int, Nφ::Int) = SphericalDirections{Float64}(Nθ, Nφ)
 Base.eltype(::Type{SphericalDirections{N}}) where {N} = Vector{N}
 Base.length(sd::SphericalDirections) = length(sd.stack)
 dim(::SphericalDirections) = 3
+isbounding(sd::SphericalDirections) = sd.Nθ > 2 && sd.Nφ > 2
 
 function Base.iterate(sd::SphericalDirections, state::Int=1)
     state == length(sd.stack)+1 && return nothing
     return (sd.stack[state], state + 1)
 end
+
+# ==================================================
+# Custom template directions
+# ==================================================
+
+"""
+    CustomDirections{N} <: AbstractDirections{N}
+
+User-defined template directions.
+
+### Fields
+
+- `directions` -- list of template directions
+- `n`          -- dimension
+- `isbounding` -- boundedness status
+
+### Notes
+
+Since custom directions may not be bounded, boundedness can either be asserted
+in the constructor or it will be determined automatically by performing an
+experiment (more precisely, we overapproximate the unit ball in the infinity
+norm using the given directions).
+
+The dimension will also be determined automatically, unless the empty vector is
+passed (in which case the optional argument `n` needs to be specified).
+"""
+struct CustomDirections{N} <: AbstractDirections{N}
+    directions::AbstractVector{<:AbstractVector{N}}
```

Comment:
```suggestion
    directions::Vector{AV}
```
where `AV<:AbstractVector{N}` is a new type parameter

Reply:
There may be applications where the directions are created dynamically, so a restriction to `Vector` may not be good.

Reply:
We decided to do the change. It can be changed again later if the need arises.